src/material/datepicker/calendar.ts

```diff
@@ -149,8 +162,21 @@ export class MatCalendarHeader<D> {
       return this._dateAdapter.getYear(date1) == this._dateAdapter.getYear(date2);
     }
     // Otherwise we are in 'multi-year' view.
-    return Math.floor(this._dateAdapter.getYear(date1) / yearsPerPage) ==
-        Math.floor(this._dateAdapter.getYear(date2) / yearsPerPage);
+    let minYear = 0;
+    if (this.calendar.minDate) {
+      minYear = this._dateAdapter.getYear(this.calendar.minDate);
+    }
+    if (this.calendar.maxDate) {
+      let maxYear = this._dateAdapter.getYear(this.calendar.maxDate);
+      minYear = maxYear - yearsPerPage + 1;
+    }
+    return Math.floor( (this._dateAdapter.getYear(date1) -  minYear) / yearsPerPage) ==
+        Math.floor( (this._dateAdapter.getYear(date2) - minYear) / yearsPerPage);
+  }
+
+  /** mod that handles case where first number is negative */
+  private _mod(a: number, b: number) {
+    return (a % b + b) % b;
```

Comment:
Since we have this function defined twice in the code, is there a common place that we can put it so that it is only defined once?